### PR TITLE
Todo application

### PR DESCRIPTION
**Pull request by poonamGawali-webonise**
Implemented todo application

### CODE REVIEW

File: pages/App/Home.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import Image from 'next/image'
+import { useState, useRef, useEffect } from 'react';
+import { FormInput } from '../../components/shared/formElements';
+import { required } from '../../utility/validation-rules';
+
+export default function Home() {
```
Comment (swapnil-webonise) [inline pages/App/Home.js]: Try to use `getStaticProps`, `getServerSideProps` methods

**Review by swapnil-webonise — CHANGES_REQUESTED** [review]

File: pages/App/Home.js (hunk 1 above)
Comment (poonamGawali-webonise) [reply pages/App/Home.js]: resolved
Added getStaticProps and getServerSideProps methods

Comment (poonamGawali-webonise) [reply pages/App/Home.js:7]: resolved
Added getStaticProps and getServerSideProps methods

**Comment by swapnil-webonise** [comment]
@poonamGawali-webonise Please lighthouse report, It should be 80+

File: package.json [hunk 2]
```diff
@@ -9,6 +9,8 @@
   },
   "dependencies": {
     "next": "10.2.3",
+    "next-compose-plugins": "^2.2.1",
```
Comment (swapnil-webonise) [inline package.json]: Remove `^` from versions, use fixed package versions.

File: pages/App/Home.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import Image from 'next/image'
+import { useState, useRef, useEffect } from 'react';
+import { FormInput } from '../../components/shared/formElements';
+import Link from 'next/link'
```
Comment (swapnil-webonise) [inline pages/App/Home.js]: Club all package imports at the top

```
import { useState, useRef, useEffect } from 'react';
import Link from 'next/link';
import Image from 'next/image';
import { FormInput } from '../../components/shared/formElements';
import { required } from '../../utility/validation-rules';
```

File: pages/App/Home.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
+import Image from 'next/image'
```
Comment (swapnil-webonise) [inline pages/App/Home.js]: Missing semi-colon

File: pages/App/Home.js (hunk 1 above)
Comment (swapnil-webonise) [reply pages/App/Home.js]: Where is `getServerSideProps` method is used?

File: pages/about.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import Image from 'next/image';
+import { playlist } from '../data/data'
+
+const About = ({playlistLists}) => {
+    return <div>
```
Comment (swapnil-webonise) [inline pages/about.js]: enclose JSX in round parenthesis

File: pages/about.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+      </ul>
+    </div>
+};
+  
+export default About;
+  
+export async function getStaticProps() {
+    return {
+        props: {
+            playlistLists: playlist
```
Comment (swapnil-webonise) [inline pages/about.js]: Missing trailing comma

File: pages/index.js [hunk 7]
```diff
@@ -1,34 +1,7 @@
-import Head from 'next/head'
-import Image from 'next/image'
-import styles from '../styles/Home.module.css'
+import App from "./App/Home";
```
Comment (swapnil-webonise) [inline pages/index.js]: Singe qoutes

File: pages/App/Home.js (hunk 4 above)
Comment (swapnil-webonise) [inline pages/App/Home.js]: Created components outside of pages directory. Because every under pages is available as a route.

**Review by swapnil-webonise — CHANGES_REQUESTED** [review]

File: data/data.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
+export const playlist = [
+    {
```
Comment (swapnil-webonise) [inline data/data.js]: The different spacing used, should be indented with 2 space tabs.

**Review by swapnil-webonise — CHANGES_REQUESTED** [review]

File: pages/App/Home.js [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+        <div className="addNotes">
+          <FormInput
+            type="text"
+            inputRef={noteElement}
+            inputClassNames="input"
+            placeholder="Enter new note"
+            inputValue={note}
+            inputName="New Note"
+            onChange={onNoteChange}
+            // validationRules={[required]}
```
Comment (swapnil-webonise) [inline pages/App/Home.js]: Remove commented code.

**Review by swapnil-webonise — CHANGES_REQUESTED** [review]

File: pages/App/Home.js (hunk 1 above)
Comment (poonamGawali-webonise) [reply pages/App/Home.js]: resolved

File: package.json (hunk 2 above)
Comment (poonamGawali-webonise) [reply package.json]: resolved

File: pages/App/Home.js (hunk 3 above)
Comment (poonamGawali-webonise) [reply pages/App/Home.js]: resolved

File: pages/App/Home.js (hunk 4 above)
Comment (poonamGawali-webonise) [reply pages/App/Home.js]: resolved

File: pages/about.js (hunk 5 above)
Comment (poonamGawali-webonise) [reply pages/about.js]: resolved

File: pages/about.js (hunk 6 above)
Comment (poonamGawali-webonise) [reply pages/about.js]: resolved

File: pages/index.js (hunk 7 above)
Comment (poonamGawali-webonise) [reply pages/index.js]: resolved

File: pages/App/Home.js (hunk 4 above)
Comment (poonamGawali-webonise) [reply pages/App/Home.js]: resolved

File: data/data.js (hunk 8 above)
Comment (poonamGawali-webonise) [reply data/data.js]: resolved

File: pages/App/Home.js (hunk 9 above)
Comment (poonamGawali-webonise) [reply pages/App/Home.js]: resolved

**Pull request merged**